gn2/wqflask/views.py

```diff
@@ -343,34 +323,35 @@ def commit_gn_doc():
     return render_template("gn_editor_results_page.html", **data)
 
 
-@app.route("/gnqna/hist/", methods=["GET"])
-@require_oauth2
-def get_hist_titles():
-    token = session_info()["user"]["token"].either(
-        lambda err: err, lambda tok: tok["access_token"])
-    response = monad_requests.get(urljoin(GN3_LOCAL_URL,
-                                          "/api/llm/hist/titles"),
-                                  headers={
-        "Authorization": f"Bearer {token}"
-    }
-    ).then(lambda resp: resp).either(
-        lambda x:  x.json(), lambda x: x.json())
-    return render_template("gnqa_search_history.html", **response)
-
-
-@app.route("/gnqna/hist/search/<search_term>", methods=["GET"])
+@app.route("/gnqna/hist", methods=["GET", "DELETE"])
 @require_oauth2
-def fetch_hist_records(search_term):
+def get_gnqa_history():
+    def _error_(resp):
+        return render_template("gnqa_errors.html",
+                               **{"status_code": resp.status_code,
+                                  **resp.json()})
     token = session_info()["user"]["token"].either(
         lambda err: err, lambda tok: tok["access_token"])
+    if request.method == "DELETE":
+        monad_requests.delete(urljoin(GN3_LOCAL_URL, "/api/llm/history"),
+                            json=dict(request.form),
```

Comment:
I thought delete requests shouldn't take in the json parameter. I don't see us explicitly define this in the `monad_requests` functions the same as we did with `put` and `post`. Also see [this comment](https://stackoverflow.com/a/32264745) for why we want to avoid this with deletes

Reply:
see this: https://requests.readthedocs.io/en/latest/_modules/requests/api/#delete   this is the  how I defined 
`it` for monads requests:https://github.com/genenetwork/genenetwork2/pull/863/commits/6db49002d4d2e69fcf4fdd6be6aceeea7b95664f